src/final_chain/final_chain.cpp

```diff
@@ -78,8 +77,8 @@ class FinalChainImpl final : public FinalChain {
   future<shared_ptr<FinalizationResult>> finalize(NewBlock new_blk,
                                                   finalize_precommit_ext precommit_ext = {}) override {
     auto p = make_shared<promise<shared_ptr<FinalizationResult>>>();
-    executor_([this, s = shared_from_this(), new_blk = move(new_blk), precommit_ext = move(precommit_ext),
```

Comment:
correct me if I am wrong, but I think this should stay there along with changes related to it

Reply:
Yeah. Strand needed exactly for that you posted above. 
I don't see a situation in which we can run finalize concurrently. Executor thread pool working on only a one thread and I think there is no reason to add some more.
Another one argument there I thought was that there is no way for some block to be processed at the same time during normal operation. But I just checked that it could be met during a sync.

So if we want that code to be safe for increasing that count of threads in a pool that logic with a strand should be returned back. But it should be reworked. Cause with a current implementation we could met with an issue that some code will try to execute after FinalChain destruction.